### DISCUSSION TITLE
PINN inverse to predict a constant in a kernel for an IDE

### DISCUSSION THREAD

**Discussion by m-ayyad** [lululxvi/deepxde#1064] [Q&A]
I am trying to do an inverse PINN to predict a parameter in the integral term in an IDE. 

     k = dde.Variable(1.0)
     c = dde.Variable(1.0)
     A = dde.Variable(1.0)

    def IDE_system(t, theta, int_mat):
       rhs = To_true / I_true * tf.math.sin(omega_true*t)

       dtheta_t = dde.grad.jacobian(theta, t)
       dtheta_tt = dde.grad.hessian(theta, t)

       lhs1 = c * tf.matmul(int_mat, dtheta_t)
       lhs2 = dtheta_tt + k * theta

       lhs = lhs1 + lhs2[: tf.size(lhs1)]

       return lhs - rhs[: tf.size(lhs1)]

     def kernel(t, tao):
        return np.exp(-A * (t - tao))

 It shows the following error

> NotImplementedError: Cannot convert a symbolic tf.Tensor (Exp:0) to a numpy array. This error may indicate that you're trying to pass a Tensor to a NumPy call, which is not supported.

I tried to change the return of the kernel to 
   `return tf.math.exp(-A * (t - tao))`
but the same error is issued

I believe the problem is in the gauss-quadrature code in the IDE soyrce code. I tried to change the it to use tensorflow instead of numpy functions but I didn't find a gauss-quadrature function in tensorflow. Did anyone face a similar problem?

Thanks,
Mahmoud

**Comment by lululxvi** [comment]
You are right that to do an inverse IDE problem, there are some codes needed to convert from numpy to tensorflow. I haven't done inverse IDE problems yet.

**Reply by m-ayyad** [reply]
thank you. I am still trying to solve it. If I solve, I will post the solution here.
